ide/static/js/signup.js

```diff
@@ -0,0 +1,298 @@
+      error_password2: 0,
+      correct: "#8d9ab0",
+      wrong: "#eb4d4b",
+      label: "#3f4a5c"
+    };
+    this.submitForm = this.submitForm.bind(this);
+    this.validateEmail = this.validateEmail.bind(this);
+    this.validateUsername = this.validateUsername.bind(this);
+    this.validatePassword1 = this.validatePassword1.bind(this);
+    this.validatePassword2 = this.validatePassword2.bind(this);
+  }
+  validateUsername() {
+    let username = document.getElementById("username-value").value;
+    let form_error_message = document.getElementById("form-error-message");
+    if (username.length > 150) {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.wrong;
+      let username_label = document.getElementById("username-label");
+      username_label.style.fontSize = "0.7em";
+      username_label.style.top = "-1em";
+      username_label.style.left = "2.4em";
+      form_error_message.innerHTML =
+        "The username can't be more than 150 characters.";
+      form_error_message.style.opacity = 1;
+      username_label.style.color = this.state.wrong;
+      this.setState({ error_username: 1 });
+    } else if (username.length == "") {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.wrong;
+      form_error_message.innerHTML = "The username can't be empty.";
+      form_error_message.style.opacity = 1;
+      let username_label = document.getElementById("username-label");
+      username_label.style.color = this.state.wrong;
+      this.setState({ error_username: 1 });
+    } else {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.correct;
+      document.getElementById("username-label").style.color = this.state.label;
+      form_error_message.style.opacity = 0;
+      this.setState({ error_username: 0 });
+    }
+  }
+  validateEmail() {
+    let email = document.getElementById("email-value").value;
+    let form_error_message = document.getElementById("form-error-message");
+    if (!email.includes("@")) {
+      document.getElementById(
+        "email-value"
+      ).style.borderColor = this.state.wrong;
+      let email_label = document.getElementById("email-label");
+      if (email != "") {
+        email_label.style.fontSize = "0.7em";
+        email_label.style.top = "-1em";
+        email_label.style.left = "2.4em";
+      }
+      form_error_message.innerHTML =
+        "The email address should include @ symbol.";
+      form_error_message.style.opacity = 1;
+      email_label.style.color = this.state.wrong;
+      this.setState({ error_email: 1 });
+    } else {
+      document.getElementById(
+        "email-value"
+      ).style.borderColor = this.state.correct;
+      document.getElementById("email-label").style.color = this.state.label;
+      form_error_message.style.opacity = 0;
+      this.setState({ error_email: 0 });
+    }
+  }
+  validatePassword1() {
```

Comment:
Rename methods based on logic in method

Reply:
But there are 2-3 checks inside the method so how do I rename it?

Reply:
based on the type of checks ex `validateEmailPassword`, etc

Reply:
Umm so if my `validatePassword` function has 2 checks for empty string and maximum char length so I should name it as `checkPasswordEmptyMaxChar`?